tests/run_tests.py

```diff
@@ -0,0 +1,69 @@
+import os
+import sys
+import argparse
+import re
+from ci_utils import xqt, pushd
+
+# Assume we are running with working directory in tests
+
+if __name__ == '__main__':
+    os.environ['WEB2PY_CONFIG'] = 'test'
+    # HINT: make sure that ``0.py`` has something like the following, that reads this environment variable:
+    #
+    # .. code:: Python
+    #   :number-lines:
+    #
+    #   config = environ.get("WEB2PY_CONFIG","production")
+    #
+    #   if config == "production":
+    #       settings.database_uri = environ["DBURL"]
+    #   elif config == "development":
+    #       settings.database_uri = environ.get("DEV_DBURL")
+    #   elif config == "test":
+    #       settings.database_uri = environ.get("TEST_DBURL")
+    #   else:
+    #       raise ValueError("unknown value for WEB2PY_CONFIG")
+
+    # HINT: make sure that you export TEST_DBURL in your environment; not set here because it's specific to local
+    # setup, possibly with a password, and thus can't be committed to the repo.
+    assert os.environ['TEST_DBURL']
+
+    # Extract the components of the DBURL. The expected format is ``postgresql://user:password@netloc/dbname``, a simplified form of the `connection URI <https://www.postgresql.org/docs/9.6/static/libpq-connect.html#LIBPQ-CONNSTRING>`_.
+    empty1, postgres_ql, pguser, pgpassword, pgnetloc, dbname, empty2 = re.split('^postgres(ql)?://(.*):(.*)@(.*)/(.*)$', os.environ['TEST_DBURL'])
+    assert (not empty1) and (not empty2)
+    os.environ['PGPASSWORD'] = pgpassword
```

Comment:
I don't understand why you don't just get both the password and the user from the TEST_DBURL?  Why make the user set more environment variables when everything we need should be there??

Reply:
This statement sets an environment variable that (on Windows) the Postgres command-line tools require to work. However, it's set based on the parsed value of `TEST_DBURL` -- the user doesn't have to set any additional command-line variables.

Or am I completely missing something?

Reply:
but the user is unconditionally set to 'postgres'

Reply:
I'm going to change it so that os.environ['PGUSER'] is set to pguser from the parsed url.

Reply:
This may be a Windows-only difference. Sorry, I just gave my first test (using Runestone!) so I'm been spending all my time on getting that working. I'll hopefully have time to think about this later today...

Reply:
Or maybe the user you set up in your URL does not have database superuser privileges??